src/jit/morph.cpp

```diff
@@ -2101,7 +2105,8 @@ GenTreePtr    Compiler::fgMakeTmpArgNode(unsigned tmpVarNum
 
         arg->gtFlags |= GTF_DONT_CSE;
```

Comment:
WEe can remove the other three places where we are setting GTF_DONT_CSE and just rely upon the one place that you just added to set this flag.